test/boost/UUID_test.cc

```diff
@@ -52,6 +52,22 @@ BOOST_AUTO_TEST_CASE(test_from_string) {
     check("e596c2f2-d29d-44a0-bb89-0a90ff928490");
     check("f28f86f5-cbc2-4526-ba25-db90c226ec6a");
     check("ce84997b-6ea2-4468-9f02-8a65abf4141a");
+
+    // shorter than 16 bytes
+    BOOST_CHECK_THROW(UUID(""), marshal_exception);
+    BOOST_CHECK_THROW(UUID("dead-beef"), marshal_exception);
+    // longer than 16 bytes
+    BOOST_CHECK_THROW(UUID("ce84997b-6ea2-4468-9f02-8a65abf4141-long-long-ago"), marshal_exception);
+    // unconvertible string
+    BOOST_CHECK_THROW(UUID("hellowol-dea2-4468-9f02-8a65abf4141a"), marshal_exception);
+    // trailing garbage in msb
+    BOOST_CHECK_THROW(UUID("ce84997b-6ea2-wxyz-9f02-8a65abf4141a"), marshal_exception);
```

Comment:
Nitpick about the comment: I'd say this is garbage, but why is it "trailing"? It's not after the correct things, it's part of the uuid, no?

Reply:
If you mean "trailing" as "in the end", you are right. I assumed that "trailing" meant "after the end", it's not exactly after. But I guess that when you parse a number, this garbage is considered after the correct part of the number. So ok.